syft/pkg/license.go

```diff
@@ -49,13 +49,18 @@ func (l Licenses) Less(i, j int) bool {
 	if l[i].Value == l[j].Value {
 		if l[i].SPDXExpression == l[j].SPDXExpression {
 			if l[i].Type == l[j].Type {
-				// While URLs and location are not exclusive fields
-				// returning true here reduces the number of swaps
-				// while keeping a consistent sort order of
-				// the order that they appear in the list initially
-				// If users in the future have preference to sorting based
-				// on the slice representation of either field we can update this code
-				return true
+				leftContents := firstTenNonEmpty(l[i].Contents)
+				rightContents := firstTenNonEmpty(l[j].Contents)
+				if leftContents == rightContents {
```

Comment:
why look at the first 10 characters? we could compare full licenses since a `string < string` comparison would bail early on the first byte difference found.

Reply:
I thought if we cut it off to first 10 non blank characters we could prevent any kind of performance issues where contents are extremely long and extremely similar.

I could compare the lengths of the strings which is pretty cheap and potentially prevent that kind of issue. 

Also happy to remove all of this and just do `string < string` if you feel it's a non issue here 😄

Reply:
we might run into issues, but most license texts are relatively short (fit within a memory page). But since there could be license texts where the first 10 chars are the same but there is a difference later in the text, so it would still be unstable.